### ISSUE TITLE
Retry can't be used with Only

### ISSUE THREAD

**Issue by xMutaGenx** [codeceptjs/CodeceptJS#1256]
#### What are you trying to achieve?

Trying to retry a scenario that has `.only` tag on it

#### What do you get instead?

The scenario only runs once.

Code:

```
Feature('Just Testing')

Scenario.only('mytest', (I) => {

    I.amOnPage('/');
    I.wait(10);
    I.see('does not exist');
}).retry(3);
```

* CodeceptJS version: 1.4.3
* NodeJS Version: 8.9.1
* Operating System: OSX

**Comment by PeterNgTr** [comment]
Scenario.only("Failing in session", (I) => {
		I.amOnPage('http://google.com');
		I.seeElement('.frfr');
}).retry(3);

`
    [1] Starting recording promises
    Emitted | suite.before ([object Object])
 › [Session] Starting singleton browser session
  Failing in session
    Emitted | test.before ([object Object])
    Emitted | test.start ([object Object])
    Emitted | step.before (I am on page "http://google.com")
    Emitted | step.after (I am on page "http://google.com")
    Emitted | step.before (I see element ".frfr")
    Emitted | step.after (I see element ".frfr")
    Emitted | step.start (I am on page "http://google.com")
    I am on page "http://google.com"
    › [Url] https://www.google.com/?gws_rd=ssl
    › [Browser:Error] Failed to load resource: net::ERR_NETWORK_CHANGED
    Emitted | step.passed (I am on page "http://google.com")
    Emitted | step.finish (I am on page "http://google.com")
    Emitted | step.start (I see element ".frfr")
    I see element ".frfr"
    [1] Error | Error
    Emitted | step.failed (I see element ".frfr")
    Emitted | step.finish (I see element ".frfr")
    [1] Error | Error
    [1] Starting <teardown> session
    Emitted | test.failed ([object Object])
    Emitted | test.finish ([object Object])
    [1] <teardown> Stopping recording promises
 › <screenshotOnFail> Test failed, saving screenshot
 › Screenshot is saving to /home/thanh/Desktop/mcmakler-ui-tests/output/Failing_in_session.failed.png
    [2] Starting recording promises
    Emitted | test.after ([object Object])
    Emitted | suite.after ([object Object])

  OK  | 0 passed   // 4s
    Emitted | global.result ([object Object])
    Emitted | global.after ([object Object])
`
Combination of only and retry resulted in passed test eventhough it is failed.

**Comment by xMutaGenx** [comment]
Hey @PeterNgTr, just to come back to this, did you test this on WebdriverIO, or Puppeteer? I was doing this on WDIO

**Comment by PeterNgTr** [comment]
I’m using puppeteer

**Comment by jancorvus** [comment]
As reported here in https://github.com/codeceptjs/CodeceptJS/issues/665#issuecomment-871133564, the issue is not only that a test passes once, but is that the result is incorrect. It shows `0 passed`

**Comment by DavertMik** [comment]
Closed in favor of #665
